src/KinesisMonologServiceProvider.php

```diff
@@ -19,12 +21,22 @@ class ServiceProvider extends LaravelServiceProvider
     public function boot()
     {
         $this->publishes([
-            __DIR__.'/../../config/kinesis.php' => config_path('kinesis.php'),
+            __DIR__ . '/../../config/kinesis.php' => config_path('kinesis.php'),
         ]);
 
-        if (config('kinesis.stream')) {
-            $monolog = Log::getLogger();
+        if ($this->app['log'] instanceof LogManager) {
+            $this->app['log']->extend('kinesis', function (Container $app, array $config) {
+                $handler = new KinesisHandler($config['stream'], $config['level'] ?? LogLevel::INFO);
 
+                return new Logger('kinesis', [$handler]);
+            });
+        }
+
+        app()->singleton(KinesisFormatter::class, function () {
```

Comment:
We don't really need a `singleton` here as Monolog stores the formatter once retrieved, so `bind` will suffice